File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.c

int lengthOfLongestSubstring(char* s) {
   int n = strlen(s);
   if(n==0)return 0;
   int l=0,r=0,max=0;
   int charset[128] = {0};
   while(r<n){
    if(charset[s[r]]==0){
        charset[s[r]]++;
        r++;
        max=max>(r-l)?max:r-l;
    }
    else{
        charset[s[l]]--;
        l++;
    }
   }
   return max;
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int l = 0;
        int max = 0;
        int n = s.length();
        if(n==0)
        return 0;

        Set<Character> set = new HashSet<>();
        for(int r = 0;r<n;r++){
            while(set.contains(s.charAt(r))){
                set.remove(s.charAt(l));
                l++;
            }
            set.add(s.charAt(r));
            max = Math.max(max, r-l+1);

        }
       
        return max;
       
    }
}